index.bs

```diff
@@ -3199,6 +3201,143 @@ The [=remote end steps=] with |command parameters| are:
 1. Return a new map matching the <code>ScriptEvaluateResultSuccess</code>
    production, with the <code>result</code> field set to |result|.
 
+#### The script.invoke Command ####  {#command-script-invoke}
+
+The <dfn export for=commands>script.invoke</dfn> command invokes a provided
+function with given arguments in a given realm.
+
+ <code>RealmInfo</code> can be either a realm or a browsing context.
+
+Issue: declare `RemoteValueArgument`.
+
+Issue: declare `LocalValueArgument`.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+      ScriptInvokeCommand = {
+        method: "script.invoke",
+        params: ScriptInvokeParameters
+      }
+
+      ScriptInvokeParameters = {
+        functionDeclaration: text;
+        arguments: [InvokeArgument];
```

Comment:
Arguably we should call this `LocalValue` since it's a value from what WebDriver calls the "local end". But also I think we need to think a bit about requirements. This setup assumes an asymmetry between the client and the browser, in that the client can refer to objects owned by the browser, but the browser can't refer to objects owned by the client. That makes some amount of sense, but let's say for sandboxed script execution you end up with some WebIDL containing a `webdriver.postMessage()` function that emits a `script.postMessage` event containing data. Then let's say, depending on the type of message you want to append to a different array on the local end (client). Obviously one way to do this is to just identify each target array with strings. But it could also be possible to write something like (on the local end, pseudocode):

```js
context.addEventListener("script.postMessage", msg => {
   {target, data} = message;
   target.append(data);
});
let fooTarget = [];
let barTarget = [];
let func = (msgName, target) => {
  window.addEventListener(msgName, event => webdriver.postMessage({target, data: event.data}));
};
context.callFunctionInSandbox(func, ["foo", fooTarget]);
context.callFunctionInSandbox(func, ["bar", barTarget]);
```

I don't know if that's a case that's worth optimising for, but it seems at least worth considering the possibility of being able to pass local-end objects through the remote.

Reply:
After a [discussion](https://github.com/w3c/webdriver-bidi/issues/140#issuecomment-942498390) decided to keep the page a single source of truth.